### PR TITLE
Enable dbt python models via serverless Dataproc and create example model

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

I had to enable Google Private Access on our default VPC, and also grant access in the prod project (BigQuery and GCS) to the staging default compute service account since local dbt runs use the staging project.

Still waiting on https://github.com/dbt-labs/dbt-bigquery/pull/636 but we can fork if necessary
~Blocked on https://github.com/dbt-labs/dbt-bigquery/issues/609~

We had a meeting today (2023-05-31) to review the tech spec, so I'd like to merge this ahead of a to-be-scheduled pairing session to test an initial model.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation
- [ ] agencies.yml

## How has this been tested?
I had created and tested an example model, but we don't actually want/need it. Pasting the source code here for posterity; it ran on our custom image provided to Dataproc.

## Screenshots (optional)
```python
import geopandas as gpd  # noqa: F401
import pandas as pd  # noqa: F401
import pyspark.sql.functions as F
import pyspark.sql.types as T
from shapely import wkt
from shapely.geometry import LineString


@F.udf(returnType=T.FloatType())
def point_array_to_linestring(arr):
    if len(arr) == 1:
        return 0
    return LineString([wkt.loads(p) for p in arr]).length


def model(dbt, session):
    dbt.config(materialized="incremental", packages=["pandas", "geopandas"])

    df = dbt.ref("dim_shapes_arrays_wkt")

    if dbt.is_incremental and session.catalog.tableExists(repr(dbt.this)):
        # only new rows compared to max in current table
        max_from_this = f"select max(_feed_valid_from) from `{dbt.this}`"
        df = df.filter(df._feed_valid_from > session.sql(max_from_this).collect()[0][0])

    df = df.withColumn(
        "length_meters", point_array_to_linestring(F.col("pt_array_wkt"))
    )

    return df
```

### CODE REVIEW

File: warehouse/.gitignore [hunk 1]
```diff
@@ -1,5 +1,8 @@
 .user.yml
 
+Miniconda3-py39_4.10.3-Linux-x86_64.sh
```
Comment (lauriemerrell) [inline warehouse/.gitignore:3]: are these files going to be created for everyone locally? they seem pretty specific for a gitignore

**Review by lauriemerrell — COMMENTED** [review]
> Still waiting on https://github.com/dbt-labs/dbt-bigquery/pull/636 but we can fork if necessary

Is this actually a blocker to merge? 

Would it be useful to add some documentation about this functionality to the warehouse README? Are there any links we could add about where to check if you need to troubleshoot any of this setup?

File: warehouse/.gitignore (hunk 1 above)
Comment (atvaccaro) [reply warehouse/.gitignore:3]: If you are building the image yes, adding to docs

**Comment by atvaccaro** [comment]
> Is this actually a blocker to merge?

No, but developer experience will be poor; I gave this caveat to tiffany and eric in our tech spec meeting yesterday.

I've added some other documentation!

File: warehouse/README.md [hunk 2]
```diff
@@ -186,6 +186,29 @@ If you prefer to install dbt locally and use your own development environment, y
...
    4. After restarting the terminal, confirm with `python3 --version` and retry `poetry install`
 
+### Dataproc configuration
+
+> If you are not using Python models or are just using the existing Dataproc configuration, you can ignore this section.
+
+[dbt docs](https://docs.getdbt.com/docs/build/python-models) exist for setting up Python models in general, as well as the specific steps required to configure BigQuery/Dataproc.
+
+The default profile template specifies `gcr.io/cal-itp-data-infra/dbt-spark:<date_tag>` as the custom image for
+Dataproc batch jobs. This image is built and pushed via the following (note that the image is hosted on Google
```
Comment (lauriemerrell) [inline warehouse/README.md:196]: nit: you start but do not close a parenthetical here

**Review by lauriemerrell — APPROVED** [review]

**Pull request merged**